feat: add AI report methods

wolfmanfp opened the pull request:
Issue: #273

andrii-bodnar commented on src/main/java/com/crowdin/client/ai/model/AiReportsFormat.java:
```diff
@@ -0,0 +1,16 @@
+
+import com.crowdin.client.core.model.EnumConverter;
+
+public enum AiReportsFormat implements EnumConverter<AiReportsFormat> {
```
Would be better to rename `AiReportsFormat` -> `AiReportFormat` for consistency

andrii-bodnar commented on src/main/java/com/crowdin/client/ai/model/AiReportDownloadLinkResponseModel.java:
```diff
@@ -0,0 +1,11 @@
+import java.util.Date;
+
+@Data
+public class AiReportDownloadLinkResponseModel {
```
Use the existing `java/com/crowdin/client/core/model/DownloadLink.java` class instead of creating a new one

andrii-bodnar commented on src/main/java/com/crowdin/client/ai/model/AiReportAttributes.java:
```diff
@@ -0,0 +1,10 @@
+
+@Data
+public class AiReportAttributes {
+    private String format;
```
Can we use the `AiReportFormat` as a type here?

andrii-bodnar reviewed (changes requested):
@wolfmanfp thank you and congrats on your first contribution!

I just left a few comments, please address them. Also, please rebase the branch, it has conflicts with the `master` branch due to the recent merge.

andrii-bodnar commented:
@wolfmanfp the CI is failing, please take a look:

```
> Task :test

AIApiTest > generateAiReportTest() FAILED
    java.lang.AssertionError at AIApiTest.java:230
```

https://github.com/crowdin/crowdin-api-client-java/actions/runs/11466729862/job/31910997760?pr=283

andrii-bodnar commented:
@wolfmanfp is there anything we can do to move this PR forward?

wolfmanfp commented:
Yes, unfortunately I'm stuck with a timezone-related bug. Specifically, when Jackson serializes the request data in the `generateAiReportTest()` method, it somehow adjusts the timezone of the two date fields to my local timezone. In TestHttpClient.java:107, this request data gets compared to the request JSON file in `resources` (`2019-09-23T11:26:54+00:00` vs. `2019-09-23T13:26:54+02:00`), and because the two strings aren't equal, the test fails. I couldn't get it to work by running the test with `-Duser.timezone=GMT` either.

I tried to disable `DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE` in JacksonJsonTransformer's `ObjectMapper`, but it needs a timezone to be set for that `ObjectMapper` so it would probably break compatibility. Even then, because of [how `SimpleDateFormat` works](https://stackoverflow.com/questions/34723683/simpledateformat-ignores-xxx-if-timezone-is-set-to-utc), it outputs `Z` instead of `+00:00`. I fixed this by writing a serializer for these two date fields which forces UTC timezone and replaces `Z` with `+00:00`, but decided against pushing it to my branch because it's not a good solution. I've replaced Date with String and it worked, but again, it's not a good solution. I don't want to delete or disable the test either.

What do you think, how should I fix this?

(Sorry for not responding, I had to write all of this down first.)

andrii-bodnar commented:
@wolfmanfp thanks for the details!

I think it would be better to remove the dependency on the `Calendar` and `Timezone` in this test and follow the approach of the existing tests.

For example:

```java
@Test
    public void updateScreenshotTest() {
        TimeZone.setDefault(tz);
        UpdateScreenshotRequest request = new UpdateScreenshotRequest();
        request.setName(name);
        request.setStorageId(storageId);
        ResponseObject<Screenshot> screenshotResponseObject = this.getScreenshotsApi().updateScreenshot(projectId, screenshotId, request);
        assertEquals(screenshotResponseObject.getData().getId(), screenshotId);
        assertEquals(new Date(119,Calendar.SEPTEMBER,23,9,35,31), screenshotResponseObject.getData().getTags().get(0).getCreatedAt());
        assertEquals(new Date(119,Calendar.SEPTEMBER,23,9,29,19), screenshotResponseObject.getData().getUpdatedAt());
    }
```

wolfmanfp commented:
Unfortunately, same problem happens as before.

andrii-bodnar commented:
Have you tried to follow the same approach as in the test case I shared in the previous comment?

wolfmanfp commented:
Yes, I did exactly what you've written.

andrii-bodnar commented:
@wolfmanfp Does the `updateScreenshotTest` also fail in your environment?

wolfmanfp commented:
@andrii-bodnar No, it doesn't.

andrii-bodnar commented:
At the moment I have no idea how to solve this serialization problem :(

It's something that probably depends on the specific environment. I would suggest adapting the test cases to pass the GH actions environment.

wolfmanfp commented:
I've changed my mind and uploaded the custom date serializer. I hope this helps :(

andrii-bodnar commented:
@wolfmanfp thanks, but I'd suggest keeping it consistent with the rest of the codebase. 

For example, we already have [`ReportsApi.java`](https://github.com/crowdin/crowdin-api-client-java/blob/master/src/main/java/com/crowdin/client/reports/ReportsApi.java) which also has these `dateFrom`, `dateTo` parameters and works without the extra serializer. 

It could be considered separately if needed, but it is out of the scope of the current PR.

wolfmanfp commented:
> For example, we already have [`ReportsApi.java`](https://github.com/crowdin/crowdin-api-client-java/blob/master/src/main/java/com/crowdin/client/reports/ReportsApi.java) which also has these `dateFrom`, `dateTo` parameters and works without the extra serializer.
Thank you for the suggestion, but those two parameters aren't used in `ReportsApiTest.java`. I didn't find any date fields used in requests in that test class.

> It's something that probably depends on the specific environment. I would suggest adapting the test cases to pass the GH actions environment.
That's what I thought when I sent this PR in, that probably the GH runner has its timezone set to GMT so it would be fine. Even then, it would crash the test because of `SimpleDateFormat`'s shenanigans (it would output `2019-09-23T11:26:54Z` instead of `2019-09-23T11:26:54+00:00`). At least I assume that's the case because the CI points me to [an HTML file](https://github.com/crowdin/crowdin-api-client-java/actions/runs/11466729862/job/31910997760?pr=283#step:5:83) which I can't access.

Meanwhile I removed the custom serializer, and found a better solution: `StdDateFormat` instead of `SimpleDateFormat`. It shouldn't break compatibility because it uses ISO 8601, doesn't screw around with timezones, and outputs `+00:00` instead of `Z`.

andrii-bodnar reviewed (approved):
@wolfmanfp thank you!
